### ISSUE TITLE
(VRR/Adaptive Sync/Freesync, etc) Auto adaptive sync functionality

### ISSUE THREAD

**Issue by orowith2os** [swaywm/sway#7093]
### Please read the following before submitting:
- We are not accepting any new window management features unless they get implemented by i3.

### Please fill out the following:
- **Description:**
Please describe in plain English what the enhancement is and what the use case is.

Currently, Sway only supports on and off options for VRR (also known as adaptive sync, which refreshes the display when new frames are available). 
Functionality to have an "auto" option for when some trigger is enabled (app goes into fullscreen on the VRR-capable display, command is ran to enable VRR) to enable VRR automatically would be appreciated.
KDE/KWin has an option like this too, which enables VRR when an app goes into fullscreen mode (having VRR enabled when an app goes into borderless would be nice, afaik it might not do this)

If this is not something Sway can implement just yet for something wlroots related, I will gladly submit an issue similar to this to the wlroots developers.

**Comment by orowith2os** [comment]
KWin merge request for VRR, it probably explains it a bit better than I can:
https://invent.kde.org/plasma/kwin/-/merge_requests/718

**Comment by emersion** [comment]
This can be implemented in a script using the IPC.

**Comment by orowith2os** [comment]
Could you be so kind as to link to sources to set up a script like that?
Having an "auto" option that uses said script would be nice to have. I get that a WM is meant to be a DIY thing, but even KWin has the auto functionality I'm talking about, and that's the one thing stopping me from swapping to Sway for daily use.
